perfkitbenchmarker/resource.py

```diff
@@ -166,6 +170,10 @@ def WaitUntilReady():
 
   def Delete(self):
     """Deletes a resource and its dependencies."""
+    if FLAGS.gce_leave_running:
```

Comment:
GCE specific stuff doesn't belong here since the resource class should be provider agnostic. If you want to leave resources running, you can simply specify --run_stage=provision,prepare,run (and omit cleanup,teardown). Let me know if this approach doesn't work for your use case.